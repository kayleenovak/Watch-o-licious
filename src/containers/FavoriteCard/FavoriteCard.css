.fav-card {
  display: flex;
  width: 500px;
  position: relative;
  margin: 30px;
}

.fav-card-background {
  position: absolute;
  height: 280px;
  width: 500px;
  top: 40px;
  bottom: 0;
  background: #1f151c;
  border-radius: 10px;
  box-shadow: 2px 2px 20px black;
}

.fav-card-image,
.fav-card-info {
  position: relative;
  z-index: 2000;
}

.episode-information {
  font-family: 'Montserrat', sans-serif;
  margin-top: 28px;
}

.episode-series {
  font-size: 14px;
}

.episode-information,
.episode-series,
.episode-runtime,
.episode-airdate {
  color: white;
}

.episode-series,
.episode-runtime,
.episode-airdate {
  font-family: 'Raleway', sans-serif;
  margin: 10px 0;
}

.fav-card-image {
  margin: 0 0 20px 25px;
  box-shadow: 2px 2px 10px black;
}

.fav-card-info {
  padding: 20px;
}

.episode-information {
  color: white;
}

.view-series-btn {
  box-shadow: 0px 1px 0px 0px #1c1b18;
  background: linear-gradient(to bottom, #aea7b9 5%, #74707a 100%);
  background-color: #aea7b9;
  border-radius: 3px;
  border: 2px solid #292929;
  display: inline-block;
  cursor: pointer;
  color: #ffffff;
  font-family: Arial;
  font-size: 14px;
  font-weight: bold;
  padding: 6px 10px;
  text-decoration: none;
  text-shadow: 1px 1px 0px black;
  margin: -5px 15px 0 0px;
}

.view-series-btn:hover {
  background: linear-gradient(to bottom, #74707a 5%, #aea7b9 100%);
  background-color: #74707a;
}

.view-series-btn:active {
  position: relative;
  top: 1px;
}

.fav-card-btns {
  box-shadow: 0px 1px 0px 0px #1c1b18;
  background: linear-gradient(to bottom, #aea7b9 5%, #74707a 100%);
  background-color: #aea7b9;
  border-radius: 3px;
  border: 2px solid #292929;
  display: inline-block;
  cursor: pointer;
  color: #ffffff;
  font-family: Arial;
  font-size: 12px;
  font-weight: bold;
  padding: 4px 8px;
  text-decoration: none;
  text-shadow: 1px 1px 0px black;
  margin: -5px 15px 5px 0px;
}

.fav-card-btns:hover {
  background: linear-gradient(to bottom, #74707a 5%, #aea7b9 100%);
  background-color: #74707a;
}

.fav-card-btns:active {
  position: relative;
  top: 1px;
}
