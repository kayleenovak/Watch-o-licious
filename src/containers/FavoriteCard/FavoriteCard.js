import React, { Component } from 'react'
import { connect } from 'react-redux'
import { addToTracked, removeFromTracked, toggleFavorite, toggleWatched, toggleWatchList } from '../../actions/index.js'
import { handleTrackedEpisode } from '../../helpers/handleToggleTracked.js'


export class FavoriteCard extends Component {

  findMatchedTvShow = () => {
    const match = this.props.tvShows.find(show => {
      return show.id === parseInt(this.props.episode.showId)
    })
    return match
  }

  render() {
    const matchedTvShow = this.findMatchedTvShow()

    return (
      <section>
        <img src={matchedTvShow.image} />
        <h3>Episode: {this.props.episode.title}</h3>
        <h5>{matchedTvShow.title}</h5>
        <button className='favorite-btn' onClick={(e) => this.props.handleTracked(e, this.props.episode, this.props.tracked)} value='favorite'>Favorite</button>
        <button className='watch-list-btn' onClick={(e) => this.props.handleTracked(e, this.props.episode, this.props.tracked)} value='watchlist'>Watch List</button>
        <button className='watched-btn' onClick={(e) => this.props.handleTracked(e, this.props.episode, this.props.tracked)} value='watched'>Watched</button>
      </section>
    )
  }
}

export const mapStateToProps = (state) => ({
  tvShows: state.tvShows,
  tracked: state.tracked
})

export const mapDispatchToProps = (dispatch) => ({
  handleTracked: (e, episode, tracked) => dispatch(handleTrackedEpisode(e, episode, tracked)) 
})

export default connect(mapStateToProps, mapDispatchToProps)(FavoriteCard)