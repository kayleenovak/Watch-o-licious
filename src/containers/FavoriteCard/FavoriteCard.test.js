import React from 'react'
import { shallow } from 'enzyme'
import { FavoriteCard, mapStateToProps, mapDispatchToProps } from './FavoriteCard'

describe('FavoriteCard', () => {
  let wrapper
  let mockHandleTracked
  let mockTvShows
  let mockTracked
  let mockEpsiode

  beforeEach(() => {
    mockHandleTracked = jest.fn()
    mockTvShows = [{
      title: 'Cake Boss', 
      id: 921,
      image: 'www.thisimgage.com',
      tracked: {
        favorites: true,
        watched: true,
        watchlist: true,
      }
    }]
    mockTracked = [{
      title: 'Cake Boss',
      image: 'www.thisimgage.com', 
      id: 921,
      tracked: {
        favorites: true,
        watched: true,
        watchlist: true,
      }
    }]
    mockEpsiode = {
      title: 'Cake Boss',
      image: 'www.thisimgage.com', 
      showId: 921,
      tracked: {
        favorites: true,
        watched: true,
        watchlist: true,
      }
    }
    wrapper = shallow(<FavoriteCard episode={ mockEpsiode } tvShows={ mockTvShows } tracked={ mockTracked } handleTracked={ mockHandleTracked }/>)
  })

  it('should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot()
  })

  it('should invoke findMatchedTvShow', () => {
    const spyMatchedTvShow = jest.spyOn(wrapper.instance(), 'findMatchedTvShow')

    wrapper.instance().forceUpdate()

    expect(spyMatchedTvShow).toHaveBeenCalled()
  })

  it('should render a section, image, h3, h5, and 3 buttons', () => {

    expect(wrapper.find('section').length).toEqual(1)
    expect(wrapper.find('img').length).toEqual(1)
    expect(wrapper.find('h3').length).toEqual(1)
    expect(wrapper.find('h5').length).toEqual(1)
    expect(wrapper.find('button').length).toEqual(3)
  })

  it('should invoke handleTracked on click of the favorite button', () => {
    const mockEvent = {
      target: {
        value: 'favorite'
      }
    }

    wrapper.find('.favorite-btn').simulate('click', mockEvent)

    expect(mockHandleTracked).toHaveBeenCalledWith(mockEvent, mockEpsiode, mockTracked)
  })

  it('should invoke handleTracked on click of the watchlist button', () => {
    const mockEvent = {
      target: {
        value: 'watchlist'
      }
    }

    wrapper.find('.watch-list-btn').simulate('click', mockEvent)

    expect(mockHandleTracked).toHaveBeenCalledWith(mockEvent, mockEpsiode, mockTracked)
  })

  it('should invoke handleTracked on click of the watched button', () => {
    const mockEvent = {
      target: {
        value: 'watched'
      }
    }

    wrapper.find('.watched-btn').simulate('click', mockEvent)

    expect(mockHandleTracked).toHaveBeenCalledWith(mockEvent, mockEpsiode, mockTracked)
  })


})