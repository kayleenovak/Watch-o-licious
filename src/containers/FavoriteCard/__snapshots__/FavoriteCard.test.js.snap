// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FavoriteCard
    episode={
      Object {
        "image": "www.thisimgage.com",
        "showId": 921,
        "title": "Cake Boss",
        "tracked": Object {
          "favorites": true,
          "watched": true,
          "watchlist": true,
        },
      }
    }
    handleTracked={[MockFunction]}
    tracked={
      Array [
        Object {
          "id": 921,
          "image": "www.thisimgage.com",
          "title": "Cake Boss",
          "tracked": Object {
            "favorites": true,
            "watched": true,
            "watchlist": true,
          },
        },
      ]
    }
    tvShows={
      Array [
        Object {
          "id": 921,
          "image": "www.thisimgage.com",
          "title": "Cake Boss",
          "tracked": Object {
            "favorites": true,
            "watched": true,
            "watchlist": true,
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          src="www.thisimgage.com"
        />,
        <h3>
          Episode: 
          Cake Boss
        </h3>,
        <h5>
          Cake Boss
        </h5>,
        <button
          className="favorite-btn"
          onClick={[Function]}
          value="favorite"
        >
          Favorite
        </button>,
        <button
          className="watch-list-btn"
          onClick={[Function]}
          value="watchlist"
        >
          Watch List
        </button>,
        <button
          className="watched-btn"
          onClick={[Function]}
          value="watched"
        >
          Watched
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "www.thisimgage.com",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Episode: ",
            "Cake Boss",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Episode: ",
          "Cake Boss",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Cake Boss",
        },
        "ref": null,
        "rendered": "Cake Boss",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorite",
          "className": "favorite-btn",
          "onClick": [Function],
          "value": "favorite",
        },
        "ref": null,
        "rendered": "Favorite",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Watch List",
          "className": "watch-list-btn",
          "onClick": [Function],
          "value": "watchlist",
        },
        "ref": null,
        "rendered": "Watch List",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Watched",
          "className": "watched-btn",
          "onClick": [Function],
          "value": "watched",
        },
        "ref": null,
        "rendered": "Watched",
        "type": "button",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            src="www.thisimgage.com"
          />,
          <h3>
            Episode: 
            Cake Boss
          </h3>,
          <h5>
            Cake Boss
          </h5>,
          <button
            className="favorite-btn"
            onClick={[Function]}
            value="favorite"
          >
            Favorite
          </button>,
          <button
            className="watch-list-btn"
            onClick={[Function]}
            value="watchlist"
          >
            Watch List
          </button>,
          <button
            className="watched-btn"
            onClick={[Function]}
            value="watched"
          >
            Watched
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "www.thisimgage.com",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Episode: ",
              "Cake Boss",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Episode: ",
            "Cake Boss",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cake Boss",
          },
          "ref": null,
          "rendered": "Cake Boss",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorite",
            "className": "favorite-btn",
            "onClick": [Function],
            "value": "favorite",
          },
          "ref": null,
          "rendered": "Favorite",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Watch List",
            "className": "watch-list-btn",
            "onClick": [Function],
            "value": "watchlist",
          },
          "ref": null,
          "rendered": "Watch List",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Watched",
            "className": "watched-btn",
            "onClick": [Function],
            "value": "watched",
          },
          "ref": null,
          "rendered": "Watched",
          "type": "button",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
